src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -244,6 +246,231 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
+            {
+                const string DiagnosticOptionPrefix = "dotnet_diagnostic.";
+                const string DiagnosticOptionSuffix = ".severity";
+
+                foreach (var (key, value) in section.Properties)
+                {
+                    // Keys are lowercased in editorconfig parsing
+                    if (key.StartsWith(DiagnosticOptionPrefix, StringComparison.Ordinal) &&
+                        key.EndsWith(DiagnosticOptionSuffix, StringComparison.Ordinal))
+                    {
+                        string diagId = key.Substring(DiagnosticOptionPrefix.Length);
+                        diagId = diagId.Substring(0, diagId.Length - DiagnosticOptionSuffix.Length);
+
+                        ReportDiagnostic? severity = null;
+                        switch (CaseInsensitiveComparison.ToLower(value))
+                        {
+                            case "default":
+                                severity = ReportDiagnostic.Default;
+                                break;
+
+                            case "error":
+                                severity = ReportDiagnostic.Error;
+                                break;
+
+                            case "warn":
+                                severity = ReportDiagnostic.Warn;
+                                break;
+
+                            case "info":
+                                severity = ReportDiagnostic.Info;
+                                break;
+
+                            case "hidden":
+                                severity = ReportDiagnostic.Hidden;
+                                break;
+
+                            case "suppress":
+                                severity = ReportDiagnostic.Suppress;
+                                break;
+
+                            default:
+                                diagnostics.Add(Diagnostic.Create(
+                                    messageProvider,
+                                    messageProvider.WRN_InvalidSeverityInAnalyzerConfig,
+                                    diagId,
+                                    value));
+                                break;
+                        }
+
+                        if (severity.HasValue)
+                        {
+                            builder[diagId] = severity.GetValueOrDefault();
+                        }
+                    }
+                }
+            }
+        }
+
+        public ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> ProcessAnalyzerConfigFiles(
+            ImmutableArray<string> analyzerConfigPaths,
+            ImmutableArray<CommandLineSourceFile> sourceFiles,
+            ref bool hadErrors,
+            DiagnosticBag diagnostics)
+        {
+            var analyzerConfigs = ArrayBuilder<EditorConfig>.GetInstance();
+
+            readEditorConfigFiles(analyzerConfigPaths, analyzerConfigs);
+
+            if (diagnostics.HasAnyErrors())
```

Comment:
Shouldn't this check be done first? The code above has no impact on the content of this `DiagnosticBag` instance.

Reply:
Ah I see now. That's really subtle as you capture some of the context (diagnostics) but pass some explicitly (analyzerConfigs). I think you should be consistent here and do one or the other. 

---
In reply to: [191616102](https://github.com/dotnet/roslyn/pull/27232#discussion_r191616102) [](ancestors = 191616102)